### DISCUSSION TITLE
shadcn's HoverCard be covered

### DISCUSSION THREAD

**Discussion by cathaysia** [TanStack/virtual#849] [Q&A] [answered]
Hi, I'm trying using this with https://ui.shadcn.com/docs/components/hover-card, however, the HoverCard be covered by next virtual item. Has anyone know how solve this?

here is my code:

```tsx
          <HoverCard>
            <HoverCardTrigger>
              <UserAvatar userName={names} className="w-4" />
            </HoverCardTrigger>
            <HoverCardContent className='z-50'>
              <div>hhhh</div>
            </HoverCardContent>
          </HoverCard>

```

Then this component are rendered as a list item in virtual.

![image](https://github.com/user-attachments/assets/85d52011-5f8f-4d8b-8a61-dddc70164d84)

**Comment by piecyk** [comment]
Thanks for reaching out! It’s hard to pinpoint the issue without a full context. If you can provide a small, working example or a minimal reproduction of the problem, I'd be happy to take a closer look.

**Reply by cathaysia** [reply]
[simple.tar.gz](https://github.com/user-attachments/files/17342931/simple.tar.gz)

Here is a simple project to reproduce this problem.

And here is the command to start it:

```console
$ npm install
$ npm dev
```

After open the browser, you need hover on `title` to show the hover-card.

**Comment by Wordllban** [comment]
I had pretty similar solution, but it was on-click popup. I assume this is caused by absolute items position
React portals helped me to solve this issue 
https://react.dev/reference/react-dom/createPortal

**Comment by cathaysia** [comment, marked as answer]
I found this problem is caused by translateY, replace it with top can solve this problem:

![image](https://github.com/user-attachments/assets/4d614cae-4afc-45f3-ac78-a8a124abd788)
![image](https://github.com/user-attachments/assets/a43ebeee-067f-475a-840b-c8d68e4d54fb)

**Comment by mfs9408** [comment]
Try wrap HoverCardPrimitive.Content with HoverCardPrimitive.Portal.

It worked for me. Something like this:

```
const HoverCardContent = React.forwardRef<
  React.ElementRef<typeof HoverCardPrimitive.Content>,
  HoverCardContent
>(
  (
    {
      className,
      animation = "data-[state=open]:animate-in data-[state=closed]:animate-out",
      align = "center",
      sideOffset = 4,
      ...props
    },
    ref,
  ) => (
    <HoverCardPrimitive.Portal>
      <HoverCardPrimitive.Content
        ref={ref}
        align={align}
        sideOffset={sideOffset}
        className={cn(
          "z-50 w-64 rounded-md border border-gray-200 bg-white p-4 text-gray-950 shadow-md outline-none data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0  data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 dark:border-gray-800 dark:bg-gray-950 dark:text-gray-50",
          animation,
          className,
        )}
        {...props}
      />
    </HoverCardPrimitive.Portal>
  ),
);
```
